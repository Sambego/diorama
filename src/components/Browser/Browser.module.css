.browser {
  width: 80vw;
  height: 80vh;
  background: var(--color-gray-7);
  border-radius: 0.5rem;
  box-shadow: 1.5rem 1.5rem 5rem 0 rgba(27, 43, 52, 0.2),
    -1.5rem -1.5rem 5rem 0 rgba(27, 43, 52, 0.2), -1.5rem 1.5rem 5rem 0 rgba(27, 43, 52, 0.2),
    1.5rem -1.5rem 5rem 0 rgba(27, 43, 52, 0.2);
}

.header {
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  padding: 1rem 2rem;
  text-align: left;
  background: var(--color-gray-medium);
  border-radius: 0.5rem 0.5rem 0 0;
}

.traffic {
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 7rem;
}

.red,
.yellow,
.green {
  align-self: center;
  display: inline-block;
  width: 1.2rem;
  height: 1.2rem;
  margin: 0 0.5rem 0 0;
  border-radius: 50%;
}

.red {
  background: var(--color-red);
}

.yellow {
  background: var(--color-yellow);
}

.green {
  background: var(--color-green);
}

.address {
  width: 100%;
  padding: 1rem 2rem;
  margin: 0 2rem 0 0;
  background: #fff;
  border: none;
  border-radius: 2rem;
  outline: 0;
  font-size: 1.6rem;
  text-align: center;
}

.iframe {
  width: 100%;
  height: calc(100% - 6rem);
}
