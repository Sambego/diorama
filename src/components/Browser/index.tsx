import React from "react";
import PropTypes from "prop-types";

import styles from "./Browser.module.css";

interface BrowserProps {
	className?: string;
	style?: React.CSSProperties;
	url: string;
}

function Browser({ style, className, url }: BrowserProps) {
	return (
		<div
			style={style}
			className={`${styles.browser} diorama-browser ${className}`}
		>
			<header className={styles.header}>
				<div className={styles.traffic}>
					<span className={styles.red} />
					<span className={styles.yellow} />
					<span className={styles.green} />
				</div>
				<input className={styles.address} type="text" value={url} disabled />
			</header>
			<iframe
				src={url}
				frameBorder="0"
				title="preview browser"
				className={styles.iframe}
			/>
		</div>
	);
}

Browser.propTypes = {
	className: PropTypes.string,
	url: PropTypes.string.isRequired,
	style: PropTypes.shape({}),
};

Browser.defaultProps = {
	className: "",
	style: {},
};

export default Browser;
