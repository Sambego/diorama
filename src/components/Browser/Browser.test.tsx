import React from "react";
import renderer from "react-test-renderer";
import Browser from ".";

it("renders correctly", () => {
	const tree = renderer.create(<Browser url="https://sambego.be" />).toJSON();
	expect(tree).toMatchSnapshot();
});
