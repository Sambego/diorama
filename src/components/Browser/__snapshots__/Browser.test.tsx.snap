// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<div
  className="_browser_1ft1l_1 diorama-browser "
  style={{}}
>
  <header
    className="_header_1ft1l_11"
  >
    <div
      className="_traffic_1ft1l_21"
    >
      <span
        className="_red_1ft1l_28"
      />
      <span
        className="_yellow_1ft1l_29"
      />
      <span
        className="_green_1ft1l_30"
      />
    </div>
    <input
      className="_address_1ft1l_51"
      disabled={true}
      type="text"
      value="https://sambego.be"
    />
  </header>
  <iframe
    className="_iframe_1ft1l_63"
    frameBorder="0"
    src="https://sambego.be"
    title="preview browser"
  />
</div>
`;
